nep-Y.mediawiki

```diff
@@ -0,0 +1,60 @@
+<pre>
+  NEP: <to be assigned>
+  Title: Contract deployment/update callback function
+  Author: Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2024-07-18
+</pre>
+
+==Abstract==
+
+This proposal standardizes the callback function that can be implemented by contracts to have some code executed right after initial deployment or update.
+
+==Motivation==
+
+Contracts may need to run some initialization/update code right after their deployment or upgrade. Neo N3 has provided a mechanism for that since version 3.0, but it was never properly standardized, so this proposal fixes it.
+
+==Specification==
+
+Neo N3 has a native <code>ContractManagement</code> contract that is used for contract deployment and update via <code>deploy</code> and <code>update</code> methods of it. After settling with the new NEF and manifest both of them can invoke a special <code>_deploy</code> method if it's implemented by the contract.
+
+===Methods===
+
+====_deploy====
+
+<pre>
+{
+  "name": "_deploy",
+  "safe": false,
+  "parameters": [
+    {
+      "name": "data",
+      "type": "Any"
+    },
+    {
+      "name": "update",
```

Comment:
Just a minor note: NeoGo has a lot of examples with `isUpdate` parameter name instead of `update` (including the native Management itself). I know it's not important for the method itself, but what is the common standard for C# contracts?

Reply:
It was this way in #154 and likely that's because of https://github.com/neo-project/neo/blob/1488958d8b39057009f645a777ab4876029fecff/src/Neo/SmartContract/Native/ContractManagement.cs#L65, so `update` is a bit more canonical and we can just change to this name. Also, see
 * https://github.com/neo-project/non-native-contracts/blob/8d72b92e5e5705d763232bcc24784ced0fb8fc87/src/NameService/NameService.cs#L440
 * https://github.com/neo-project/neo-devpack-dotnet/blob/8f4cd2ba3de4e54946a563d4a1e5e937a31235d8/src/Neo.SmartContract.Template/templates/neocontractowner/Ownable.cs#L59
 * https://github.com/neo-project/neo-devpack-dotnet/blob/8f4cd2ba3de4e54946a563d4a1e5e937a31235d8/src/Neo.SmartContract.Template/templates/neocontractnep17/Nep17Contract.cs#L90